Some minor bug fixes plus some new list functions

robert-dodier opened the pull request:
I've made some changes which make it possible to load Mixima into a ANSI-Common Lisp compliant Lisp implementation. Also, I've added implementations of some more list functions (MemberQ, PadLeft/PadRight, RotateLeft/RotateRight). Finally, I also changed the implementation of NestList(f, x, n) so that it returns a list of length n + 1, with the first element being x; this appears to conform to the description in the Mathematica documentation.

This version of Mixima can translate and run the prime testing via Chebyshev polynomials which is described in this MathOverflow post: https://mathoverflow.net/questions/286304/chebyshev-polynomials-of-the-first-kind-and-primality-testing

Commit:
Fix minor syntax errors.
 * compatibility/derivatives_integrals.lisp: ensure balanced parentheses.
 * translator/maxima.lisp: ensure balanced parentheses;
   also, replace invalid construct ('expr) with just expr
   (I'm guessing the intent is to return expr unmodified).

Commit:
In definition of macro DEFMIX, restate nonstandard construct,
with extra arguments after "&rest body", as LET.

Commit:
MAPPLY takes 3 arguments, not 2. (The third argument is the same as the first.)

Commit:
Let the interpreter for scripts be SBCL instead of GCL.

Commit:
New list functions: MemberQ, PadLeft, PadRight, RotateLeft, RotateRight.
For PadLeft, PadRight, RotateLeft, and RotateRight, only a single integer argument
is allowed; more general arguments are not implemented yet.

Commit:
Change NestList(f, x, n) so that it returns a list of n + 1 elements,
with the first element being x, and the second and later being f(x), f(f(x)), etc.
I believe this is the behavior described by the Mathematica documentation:
https://reference.wolfram.com/language/ref/NestList.html

jlapeyre commented:
Thanks very much Robert! Sorry for the delay. I'm busy with a move and new job. Will get to this ASAP.

robert-dodier commented:
Great, John, no hurry, glad to hear you got the message.

robert-dodier committed:
Various updates to get load("compatibility/mixima.mac") to succeed.

Some functionality seems to work now, although

  load ("compatibility/mixima.mac");
  load ("translator/mockmma-mixima-shell.lisp");
  entermma (0);

doesn't behave as expected -- I can't seem to get it to evaluate inputs,
and for some inputs it says ''There is no package with name "maxima"''.

Changes in this commit:

 * BUGS: mention a couple of bugs found recently

 * compatibility/devel.mac: omit files which have disappeared

 * compatibility/general.mac: define functions Re and Im
   (might consider moving those into another file)

 * compatibility/mixdoc.in.lisp: rename to compatibility/mixdoc.lisp

 * compatibility/mixima.mac: simplify business about setting path and loading files;
   omit files which have disappeared

 * translator/eval.lisp: (1) try to simplify stuff around error handling via MAXIMA::ERRSET
   (doesn't appear to be working correctly yet);
   (2) replace LISP:FOO with CL:FOO

 * translator/maxima.lisp: (1) MAXIMA::MAPPLY need a third operand;
   (2) replace USER::FOO with CL:FOO

 * translator/mixima-mockmma-parser.lisp: print debugging info before throwing error

 * translator/mma.lisp: simplify stuff about path

 * translator/mockmma-mixima-shell.lisp: cut out commented-out code;
   simplify stuff about path; call MAXIMA::$LOAD instead of LOAD;
   try to simplify stuff around error handling via MAXIMA::ERRSET

robert-dodier committed:
Ensure that all calls to MAPPLY have three arguments.

robert-dodier committed:
set_file_search_paths.mac: new script to set file search paths.

jlapeyre commented:
Thanks very much Robert. I forgot about this. I haven't been engaged with CL for the past few years.
